packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -327,13 +334,34 @@ abstract class BindingBase {
     registerServiceExtension(
       name: name,
       callback: (Map<String, String> parameters) async {
-        if (parameters.containsKey(name))
-          await setter(double.parse(parameters[name]));
+        if (parameters.containsKey(name)) {
+          final double value = double.parse(parameters[name]);
+          await setter(value);
+          _postExtensionChangedEvent(name, value);
```

Comment:
Here's an idea: pass back the value returned by awaiting the
getter().
That should keep the state exactly identical with what happens if we call the getter to get the state and keep code on the clients completely regular.

Reply:
Done.